File: Leetcode/137. Single Number II.cpp
class Solution {
public:
    int singleNumber(vector<int>& nums) {
        long long int ans = 0, two = 1,temp = 0, n;
        n = nums.size();
        for(int i = 1;i <= 32; i++){
            temp = 0;
            for(int j = 0 ; j<n; j++)
            temp += ((nums[j] & two)>0);
            ans += (two * (temp%3));
            two <<= 1;
        }
        
        return (int)ans;
        
    }
};

File: Leetcode/137. Single Number II.java
https://leetcode.com/problems/single-number-ii
class Solution {
    public int singleNumber(int[] nums) {
        int ans = 0;
        int n = nums.length;
        int two = 1;
        for(int i=1;i<=32;i++){
            int ct = 0;
            for(int j=0;j<n;j++){
                if((nums[j]&two) !=0){
                    ct++;
                }
            }
            ans += (ct%3 !=0) ? two:0;
            two <<=1;
        }
        return ans;
    }
}
